File: dotnet/Allmark/Render/HtmlBlockRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class HtmlBlockRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "html_block",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        state.Output.Append(node.Content);
    }
}

File: zig/src/render/htmlBlockRenderer.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;

pub fn render(node: *const MarkdownNode, state: *RendererState, decode: ?bool) void {
    _ = decode;
    state.output.appendSlice(state.allocator, node.content) catch unreachable;
}

pub const htmlBlockRenderer = Renderer{
    .name = "html_block",
    .render = render,
};
